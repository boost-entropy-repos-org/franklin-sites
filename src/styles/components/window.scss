@import "../settings";
@import "../colours";

.window {
  display: flex;
  flex-direction: column;
  border: 0.1rem solid $colour-uniprot-blue;

  &--with-shadow {
    box-shadow: 0px 0px 10px 2px rgba($colour-uniprot-blue,0.5)
  }

  &__header {
    width: 100%;
    height: 2rem;
    color: $colour-white;
    background-color: $colour-uniprot-blue;
    font-weight: 700;
    padding: 0 $global-padding / 2;
    display: flex;
    line-height: 2rem;
    justify-content: center;
    align-items: center;

    &__title {
      margin-right: auto;
    }
  }

  &__content {
    width: 100%;
    flex-grow: 1;
    padding: $global-padding / 2;
    background-color: $colour-white;
  }

  &__footer {
    width: 100%;
    background-color: $colour-seashell-grey;
    display: flex;
    justify-content: flex-end;
    padding: $global-padding / 2;

    .button {
      margin-bottom: 0;
      margin-left: $global-padding / 2;
      padding: 0.5rem 1rem;
    }
  }

  &__default-close-button {
    margin-left: auto;

    &--icon-only {
      width: 1rem;
      height: 1rem;
      background-color: $colour-white;
      color: $colour-uniprot-blue;
      cursor: pointer;
      border-radius: 0.5rem;
    }
  }
}
