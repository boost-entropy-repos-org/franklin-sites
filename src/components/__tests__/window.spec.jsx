import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import Window from '../window/window';
import { WindowActionButton } from '../window/window-buttons';

const customActionButtons = [
  <WindowActionButton
    text="No"
    key="window-action-no"
    onClick={() => null}
  />,
  <WindowActionButton
    text="Yes"
    key="window-action-yes"
    onClick={() => null}
    primary
  />,
];

describe('Window component', () => {
  test('should render', () => {
    const component = renderer
      .create(
        <Router>
          <Window
            width="20rem"
            height="15rem"
            title="Alert"
            withHeaderCloseButton
            withFooterCloseButton
            onWindowOpen={() => null}
            onWindowClose={() => null}
            withShadow
            key="full-featured-window"
          >
            Hello, World!
          </Window>
        </Router>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('should render with custom action buttons', () => {
    const component = renderer
      .create(
        <Router>
          <Window
            width="20rem"
            height="10rem"
            title="Confirm"
            actionButtons={customActionButtons}
            key="basic-dialog-window"
          >
            Complete this action?
          </Window>
        </Router>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
