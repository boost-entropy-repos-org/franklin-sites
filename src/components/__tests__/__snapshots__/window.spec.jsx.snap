// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Window component should render 1`] = `
<div
  className="window window--with-shadow"
  style={
    Object {
      "height": "15rem",
      "left": "calc((100vw - 20rem) / 2)",
      "top": "calc((80vh - 15rem) / 2)",
      "width": "20rem",
    }
  }
>
  <div
    className="window__header"
  >
    <span
      className="window__header__title"
    >
      Alert
    </span>
    <button
      className="window__action-button window__action-button--icon-only window__default-close-button window__default-close-button--icon-only"
      onClick={[Function]}
      type="button"
    >
      <test-file-stub
        height="10"
        width="10"
      />
    </button>
  </div>
  <div
    className="window__content"
  >
    Hello, World!
  </div>
  <div
    className="window__footer"
  >
    <button
      className="window__action-button window__action-button--icon-only button window__default-close-button"
      onClick={[Function]}
      type="button"
    >
      Close
    </button>
  </div>
</div>
`;

exports[`Window component should render with custom action buttons 1`] = `
<div
  className="window"
  style={
    Object {
      "height": "10rem",
      "left": "calc((100vw - 20rem) / 2)",
      "top": "calc((80vh - 10rem) / 2)",
      "width": "20rem",
    }
  }
>
  <div
    className="window__header"
  >
    <span
      className="window__header__title"
    >
      Confirm
    </span>
  </div>
  <div
    className="window__content"
  >
    Complete this action?
  </div>
  <div
    className="window__footer"
  >
    <button
      className="window__action-button window__action-button--icon-only button secondary"
      onClick={[Function]}
      type="button"
    >
      No
    </button>
    <button
      className="window__action-button window__action-button--icon-only button"
      onClick={[Function]}
      type="button"
    >
      Yes
    </button>
  </div>
</div>
`;
